src/Tempest/Support/src/ArrayHelper.php

```diff
@@ -768,12 +768,85 @@ public function mapTo(string $to): self
     }
 
     /**
-     * Sorts the array in ascending order.
+     * Returns a new instance of this array sorted by its values.
+     *
+     * @param bool $desc Sorts in descending order if `true`; defaults to `false` (ascending).
+     * @param bool|null $preserveKeys Preserves array keys if `true`; reindexes numerically if `false`.
+     *                                Defaults to `null`, which auto-detects preservation based on array type  (associative or list).
+     * @param int $flags Sorting flags to define comparison behavior, defaulting to `SORT_REGULAR`.
+     * @return ($preserveKeys is true ? self<TKey, TValue> : self<int|TKey, TValue>)
```

Comment:
I don't think IDEs can properly parse this syntax.

Then why not go with something like `@return self<array-key, TValue> Key type depends on $preserveKeys` or something similar ?
Like that, the IDE still works with `array-key` syntax, we don't have the `TKey` because we aren't sure that this is the same as the current ones ( since preserveKeys can change them ). But we make a text explanations for developers ( a more precise explanations can be done, but I think $preserveKeys is self-explanatory enough ).

However, If there is a syntax that allow to conditionally set return type based on a parameter, we could change them to it.
Maybe it's possible with Generics but I'm not experienced enough with them to know how.

Reply:
It should work, i used that syntax from: https://phpstan.org/writing-php-code/phpdoc-types#conditional-return-types
PHPstorm did not have any troubles with it, not sure about other IDE's though. 

But the problem is that i cannot rely on the `$preserveKeys` variable since the default is null and in that case the method will determine to preserve the keys based on the `isAssoc` method.

So yeah i think using  `@return self<array-key, TValue>` with a proper comment might be better.

Reply:
Thanks for the link 🙌 

I see, so yes I'm not sure that can be based on in-method code but only on parameters.

Reply:
Hehe it only works with parameters, i already tried that 😉